### DISCUSSION TITLE
How do I stop HDD activity every ~5 second on hdd - TrueNAS SCALE 22.02.4 with ZFS?

### DISCUSSION THREAD

**Discussion by Dozelr** [openzfs/zfs#14421] [Q&A] [answered]
I have TrueNAS SCALE 22.02.4 with Jellyfin. System Dataset Pool is on boot-pool - SSD drive, but sound from my Toshiba N300 12TB is still on every 5 seconds. I just wanna keep my Jellyfin and that's it. Constant sound from HDD is driving me and my family crazy, server is in another room, but sound from HDD writes/reads is very loud, especially at night ~46dB. No one seems to know how to solve this issue. It is probably flushing of ZFS TXGs to the underlying storage media. Is there any solution for stopping it?

**Comment by amotin** [comment]
5 seconds indeed sounds like a maximum TXG commit timeout, used if pool _is_ minimally modified.  If atime property is enabled (default), it may even be caused by read accesses.  If disabled (should usually be possible), then it must be some write.  If you are saying the system dataset is on a boot poll, then it should not be the TrueNAS itself, but something that is running on top, like the Jellyfin.  Just find what is writing. If nothing else, may be you could add another small SSD pool to mount it over the place that is getting the writes to remove them from the HDD pool.

**Comment by Dozelr** [comment]
Ok, so maybe another angle. Anyone know where is zfs.conf in TrueNAS SCALE 22.02.4? It is not in /etc/modprobe.d/ Where can I change zfs_txg_timeout=300?

**Comment by IvanVolosyuk** [comment, marked as answer]
AFAIK, this can be changed even after the zfs module is loaded. On linux I
simply do in one of startup scripts:

echo 120 > /sys/module/zfs/parameters/zfs_txg_timeout

**Comment by Dozelr** [comment]
Thank you Ivan. I have tested echo 5000 at night, and now I'm using script with echo 600 >/sys/module/zfs/parameters/zfs_txg_timeout

It is a huge difference. Nevertheless I'm working on a hardware solution at my workshop.

**Reply by IvanVolosyuk** [reply]
One downside I have in my setup with zfs_txg_time = 120 and sync=disabled is that hard drivers by default spin down due to inactivity and s.m.a.r.t. Load_Cycle_Count increases almost every 2 minutes, which is probably bad for the drive health long term. Thus, I have to tweak the power management and timeout for my drives.

**Reply by Dozelr** [reply]
You are right, spindowns of hdds are bad for them. On TrueNAS Scale I have my hdd standby - always on, the power saving options with spindown are off (by default). I don't think the echo 600 should change that - correct me if I'm wrong.  

My server is taking from the wall 21W - 22W in idle and the same with 2 TVs using Jellyfin - I was going for low power consumption. I will look on the power consumption after few hours of idle. In my situation less than 21W = hdd spindown. 

On Ubuntu I had ~15W. I think then hdd was in idle mode.

**Comment by jxdking** [comment]
It means there are disk activities going on. Even with 5 seconds zfx_txg_timeout, my hard drives would spin down after 20 minutes in idle. You probably need to check what is constantly writing data.
